src/main/scala/firrtl/RenameMap.scala

```diff
@@ -178,13 +178,151 @@ final class RenameMap private () {
   private def completeGet(key: CompleteTarget): Option[Seq[CompleteTarget]] = {
     val errors = mutable.ArrayBuffer[String]()
     val ret = if(hasChanges) {
-      val ret = recursiveGet(mutable.LinkedHashSet.empty[CompleteTarget], errors)(key)
+      val ret = recursiveGet(errors)(key)
       if(errors.nonEmpty) { throw IllegalRenameException(errors.mkString("\n")) }
       if(ret.size == 1 && ret.head == key) { None } else { Some(ret) }
     } else { None }
     ret
   }
 
+
+  /** Checks for renames of only the component portion of a [[ReferenceTarget]]
+    * NOTE: assumes key has been normalized
+    * @param errors Used to record illegal renames
+    * @param key Target to rename
+    * @return Renamed targets
+    */
+  private def componentGet(errors: mutable.ArrayBuffer[String])(key: ReferenceTarget): Option[Seq[CompleteTarget]] = {
+    def traverseTokens(key: ReferenceTarget): Option[Seq[CompleteTarget]] = {
+      if (underlying.contains(key)) {
+        Some(underlying(key))
+      } else {
+        key match {
+          case t: ReferenceTarget if t.component.nonEmpty =>
+            val last = t.component.last
+            val parent = t.copy(component = t.component.dropRight(1))
+            traverseTokens(parent).map(_.flatMap { x =>
+              (x, last) match {
+                case (t2: ReferenceTarget, Field(f)) => Some(t2.field(f))
+                case (t2: ReferenceTarget, Index(i)) => Some(t2.index(i))
```

Comment:
I think this is not allowed according to this test case? https://github.com/freechipsproject/firrtl/pull/967/files#diff-d4b2eb4d442e9245d392a91f972f5a5aR278
It would be easy to allow this case though.

Reply:
there is also another case that I think I'm missing. for rename map `~Top|Top/i:Inline>b -> ~Top|Top>i_b` should we be able to rename `~Top|Top>i.b` to `~Top|Top>i_b`? It seems like something we should support since we allow renaming of references to instances.

Reply:
i think we need `case (t2: InstanceTarget, Field(f)) => Some(t2.ref(f))` here too? for the case where the base reference gets renamed to an instance.

and I don't think we need the `~Top|Top>foo.bar -> ~Top|Foo>bar` case. I think it makes things more clear to only allow renaming from `IsComponent` to `IsComponent` and `IsModule` to `IsModule`.

Reply:
> and I don't think we need the `~Top|Top>foo.bar -> ~Top|Foo>bar` case. I think it makes things more clear to only allow renaming from `IsComponent` to `IsComponent` and `IsModule` to `IsModule`.
Yeah, I agree with you that this tradeoff is worth it. Let's not support that case.

> rename map: `~Top|Top/i:Inline>b -> ~Top|Top>i_b`, rename `~Top|Top>i.b` to `~Top|Top>i_b`?
I agree it is difficult because we don't know the `OfModule`. I think we have to punt on it and force the user to add `~Top|Top>i.b -> ~Top|Top>i_b` to the rename map.